src/server/pkg/collection/transaction.go

```diff
@@ -243,13 +258,17 @@ func (s *stm) DelAll(prefix string) {
 }
 
 func (s *stm) Rev(key string) int64 {
+	s.Lock()
+	defer s.Unlock()
 	if resp := s.fetch(key); resp != nil && len(resp.Kvs) != 0 {
 		return resp.Kvs[0].ModRevision
 	}
 	return 0
 }
 
 func (s *stm) commit() *v3.TxnResponse {
+	s.Lock()
+	defer s.Unlock()
```

Comment:
I don't think `commit` needs a lock if `reset` doesn't either - these are both just used in `runSTM`, which is only used by `NewSTM`, so there should be no concurrent access of the object at this point.